### PR TITLE
IA-1590 org units perfs

### PR DESCRIPTION
**Pull request by beygorghor**
We have too much renders on the org unit search page.
To reproduce, go locally on orgunits search page, set Geographical data filter to only points.
Search and try to select a row in the multi select, it’s very slow, on Axel Computer it can take a few seconds !!

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests

## Changes
- split the map into another component to avoid re-render of the map on each user action
- implement useMemo to avoid rows calculation
- split cell into separate components.

## How to test

Explain how to test your PR.
Reproduce same search

### CODE REVIEW

File: hat/assets/js/apps/Iaso/domains/orgUnits/index.tsx [hunk 1]
```diff
@@ -164,34 +146,22 @@ export const OrgUnits: FunctionComponent<Props> = ({ params }) => {
...
-            setSelection(newSelection);
+    const getSearchColor = useCallback(
+        currentSearchIndex => {
+            const currentSearch = searches[currentSearchIndex];
+            let currentColor;
+            if (currentSearch) {
+                currentColor = currentSearch.color;
+            }
+            if (!currentColor) {
+                currentColor = getChipColors(0);
```
Comment (olethanh) [inline hat/assets/js/apps/Iaso/domains/orgUnits/index.tsx]: shouldn't it be getChipColors(currentSearchIndex) instead of 0?

**Review by olethanh — APPROVED** [review]

**Comment by olethanh** [comment]
Haven't reviewed the whole code but tested it in the frontend and it seems to works well, except sometimes it lost the source parametter but apparently that's a known bug

**Comment by codeKameleon** [comment]
still need to go through the whole code but works well ! I get huge perf improvements locally

File: hat/assets/js/apps/Iaso/domains/orgUnits/index.tsx [hunk 2]
```diff
@@ -200,7 +170,7 @@ export const OrgUnits: FunctionComponent<Props> = ({ params }) => {
 
     const onSearch = useCallback(
         newParams => {
-            handleTableSelection('reset');
+            // handleTableSelection('reset');
```
Comment (codeKameleon) [inline hat/assets/js/apps/Iaso/domains/orgUnits/index.tsx:173]: remove the comment ?

**Review by codeKameleon — APPROVED** [review]

**Pull request merged**